apps/designer/app/canvas/shared/use-shortcuts.ts

```diff
@@ -1,38 +1,59 @@
 import { useHotkeys } from "react-hotkeys-hook";
 import store from "immerhin";
-import {
-  type Instance,
-  publish,
-  useSubscribe,
-  components,
-} from "@webstudio-is/react-sdk";
+import { type Instance, components } from "@webstudio-is/react-sdk";
 import { shortcuts, options } from "~/shared/shortcuts";
+import { publish, useSubscribe } from "~/shared/pubsub";
 import { useSelectedInstance } from "./nano-states";
 import { copy, paste } from "./copy-paste";
 import {
   useRootInstance,
   useTextEditingInstanceId,
 } from "~/shared/nano-states";
+import { type SelectedInstanceData } from "~/shared/canvas-components";
+
+declare module "~/shared/pubsub" {
+  export interface PubsubMap {
+    cancelCurrentDrag: undefined;
+    deleteInstance: {
+      id: Instance["id"];
+    };
+    openBreakpointsMenu: undefined;
+    selectBreakpointFromShortcut: number;
+    selectInstance?: SelectedInstanceData;
+    togglePreviewMode: undefined;
+    zoom: "zoomOut" | "zoomIn";
+  }
+}
 
 const inputTags = ["INPUT", "SELECT", "TEXTAREA"] as const;
 
 type HandlerEvent = {
-  key: string;
+  key?: string;
   preventDefault?: () => void;
 };
 
 const togglePreviewMode = () => {
-  publish<"togglePreviewMode">({ type: "togglePreviewMode" });
+  publish({ type: "togglePreviewMode" });
 };
 
 const publishSelectBreakpoint = ({ key }: HandlerEvent) => {
+  if (!key) {
+    throw new Error(
+      "`publishSelectBreakpoint` doesn't account for being called without a `key`"
+    );
+  }
```

Comment:
This is not ideal but I've felt that it's better to add such a throw here, just in case. The problem is that [this file](https://github.com/webstudio-is/webstudio-designer/blob/b913ac73d726ad41ca33bda2d44f1bd6fdf09ea9/apps/designer/app/designer/features/topbar/menu/menu.tsx) piggybacks on the shortcut events - but in there we don't have access (or at least not always) to the `key`.

Perhaps this file shouldn't do it at all and it should just use its own, explicit, events - but that would be a separate refactor.

Not all shortcut events are actually dispatched from that `menu.tsx` and those to which I've added those throws arent dispatched from there. I can't easily filter them out though because it's a list that is being looped on etc. So I've just made `HandlerEvent["key"]` optional for the time-being

Reply:
I think its better to make sure the key is always provided instead.

Reply:
Also key could totally stay a string, the other number type should change, dom key is string

Reply:
> I think its better to make sure the key is always provided instead.

In such a case - do you want to provide fake keys to those dispatched `shortcut` events? Or do you want to refactor them to explicit events (instead of piggybacking on the existing `shortcut` event)?